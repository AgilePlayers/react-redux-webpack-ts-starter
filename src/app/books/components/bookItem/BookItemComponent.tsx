import React from 'react';
import './bookItem.scss';

export interface BookItemComponentProps {
  info: {
    title: string;
    description: string;
    imageLinks: {
      smallThumbnail: string;
    };
  };
}

const BookItemComponent: React.SFC<BookItemComponentProps> = ({ info }) => {
  const {
    title,
    description,
    imageLinks: { smallThumbnail },
  } = info;

  return (
    <li className="book-item">
      <div className="book-item-thumbnail">
        <img src={smallThumbnail} alt="thumbnail" />
      </div>
      <div className="book-item-details">
        <div className="book-item-title">{title}</div>
        <div className="book-item-description">{description}</div>
      </div>
      <div className="book-item-clear" />
    </li>
  );
};

export default BookItemComponent;
