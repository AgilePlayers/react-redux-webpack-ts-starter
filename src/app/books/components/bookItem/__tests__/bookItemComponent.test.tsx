import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import BookItemComponent from '../BookItemComponent';

describe('books / bookItem / BookItemComponent', () => {
  let component: ShallowWrapper;
  let props;

  beforeEach(() => {
    props = {
      info: {
        title: 'Hello Poland!',
        description: 'Hello Poland description',
        imageLinks: {
          smallThumbnail: 'image-mock',
        },
      },
    };
    component = shallow(<BookItemComponent {...props} />);
  });
  test('should render', () => {
    expect(component.length).toEqual(1);
  });

  test('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  test('should contains elements with book-item classes', () => {
    expect(component.find('li.book-item').length).toEqual(1);
    expect(component.find('div.book-item-thumbnail').length).toEqual(1);
    expect(component.find('div.book-item-details').length).toEqual(1);
    expect(component.find('div.book-item-title').length).toEqual(1);
    expect(component.find('div.book-item-description').length).toEqual(1);
    expect(component.find('div.book-item-clear').length).toEqual(1);
  });
});
