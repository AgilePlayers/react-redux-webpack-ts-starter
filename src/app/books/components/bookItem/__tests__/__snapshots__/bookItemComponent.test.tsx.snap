// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`books / bookItem / BookItemComponent should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BookItemComponent
    info={
      Object {
        "description": "Hello Poland description",
        "imageLinks": Object {
          "smallThumbnail": "image-mock",
        },
        "title": "Hello Poland!",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="book-item-thumbnail"
        >
          <img
            alt="thumbnail"
            src="image-mock"
          />
        </div>,
        <div
          className="book-item-details"
        >
          <div
            className="book-item-title"
          >
            Hello Poland!
          </div>
          <div
            className="book-item-description"
          >
            Hello Poland description
          </div>
        </div>,
        <div
          className="book-item-clear"
        />,
      ],
      "className": "book-item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="thumbnail"
            src="image-mock"
          />,
          "className": "book-item-thumbnail",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "thumbnail",
            "src": "image-mock",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="book-item-title"
            >
              Hello Poland!
            </div>,
            <div
              className="book-item-description"
            >
              Hello Poland description
            </div>,
          ],
          "className": "book-item-details",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Hello Poland!",
              "className": "book-item-title",
            },
            "ref": null,
            "rendered": "Hello Poland!",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Hello Poland description",
              "className": "book-item-description",
            },
            "ref": null,
            "rendered": "Hello Poland description",
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "book-item-clear",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="book-item-thumbnail"
          >
            <img
              alt="thumbnail"
              src="image-mock"
            />
          </div>,
          <div
            className="book-item-details"
          >
            <div
              className="book-item-title"
            >
              Hello Poland!
            </div>
            <div
              className="book-item-description"
            >
              Hello Poland description
            </div>
          </div>,
          <div
            className="book-item-clear"
          />,
        ],
        "className": "book-item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="thumbnail"
              src="image-mock"
            />,
            "className": "book-item-thumbnail",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "thumbnail",
              "src": "image-mock",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="book-item-title"
              >
                Hello Poland!
              </div>,
              <div
                className="book-item-description"
              >
                Hello Poland description
              </div>,
            ],
            "className": "book-item-details",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Hello Poland!",
                "className": "book-item-title",
              },
              "ref": null,
              "rendered": "Hello Poland!",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Hello Poland description",
                "className": "book-item-description",
              },
              "ref": null,
              "rendered": "Hello Poland description",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "book-item-clear",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
